clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -461,13 +461,73 @@ check_function_markers()
     return true;
 }
 
+bool
+check_repeated_syscall_with_same_pc()
+{
+    constexpr addr_t ADDR_ONE = 0x7fcf3b9dd9e9;
+    // Negative: syscalls with the same PC.
+#if defined(X86_64) || defined(X86_32) || defined(ARM_64)
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(1, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+#    if defined(X86_64) || defined(X86_32)
+            gen_instr_encoded(ADDR_ONE, { 0x0f, 0x05 }), // 0x7fcf3b9dd9e9: 0f 05 syscall
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 0),
+            gen_marker(1, TRACE_MARKER_TYPE_CPU_ID, 3),
+            gen_instr_encoded(ADDR_ONE, { 0x0f, 0x05 }), // 0x7fcf3b9dd9e9: 0f 05 syscall
+#    elif defined(ARM_64)
+            gen_instr_encoded(ADDR_ONE,
+                              0xd4000001), // 0x7fcf3b9dd9e9: 0xd4000001 svc #0x0
+            gen_marker(1, TRACE_MARKER_TYPE_TIMESTAMP, 0),
+            gen_marker(1, TRACE_MARKER_TYPE_CPU_ID, 3),
+            gen_instr_encoded(ADDR_ONE,
+                              0xd4000001), // 0x7fcf3b9dd9e9: 0xd4000001 svc #0x0
+#    else
+        // TODO i#5871: Add AArch32 (and RISC-V) encodings.
+#    endif
+        };
+        if (!run_checker(memrefs, true, 1, 5, "Repeated syscall instrs with the same PC",
```

Comment:
I think it'll be useful if run_checker also verifies where the error occurred. For now, please add a TODO.

Reply:
Do you mean at which ref count the error occurred?  That's already checked: that's the `5` parameter here.